isthmus/src/main/java/io/substrait/isthmus/SubstraitToSql.java

```diff
@@ -66,16 +66,22 @@ public RelNode substraitRelToCalciteRel(
     return SubstraitRelNodeConverter.convert(relRoot, relOptCluster, catalogReader, parserConfig);
   }
 
-  private static final SqlDialect DEFAULT_DIALECT =
-      new SqlDialect(
-          SqlDialect.EMPTY_CONTEXT
-              .withDatabaseProduct(SqlDialect.DatabaseProduct.SNOWFLAKE)
-              .withDatabaseProductName("SNOWFLAKE")
-              .withIdentifierQuoteString(null)
-              .withNullCollation(NullCollation.HIGH));
+  private static final SqlDialect.Context DEFAULT_DIALECT_CONTEXT =
```

Comment:
Just curious on what the implications of these defaults are. Does this determine whether column name references are double quoted? What about the case sensitive case? A detail, but eventually we'll need to deal with that. Also, with SQL dialect set to SNOWFLAKE, does `supportsApproxCountDistinct()` still return false? That seems like a bug, no? Maybe some comments would be good here.

Reply:
`.withIdentifierQuoteString` will determine how to quote an identifier. Passing a `null` means no quoted identifier. Passing `"` would double quote all identifiers, including table/column names. 

Calcite has a `SnowflakeDialect`[1]. It does set `supportsApproxCountDistinct` to true. However, it specifies double quote for all identifier, which would lead problem to table/column searching in catalog. 

I changed the code to just use Calcite's `EMPTY_CONTEXT` to construct a sql dilect, since `EMPTY_CONTEXT` defaults to use null as IdentifierQuoteString. Same as SnowflakeDialect, I set `supportsApproxCountDistinct` for the constructed sql dialect. 

1. https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java#L28-L29